### PR TITLE
Feature/stable api key

### PR DESCRIPTION
**Pull request by agrav**

### CODE REVIEW

File: src/pytailor/clients/auth.py [hunk 1]
```diff
@@ -1,15 +1,111 @@
...
+    # refresh/load tokens with the following efforts in order:
+    # 1. load tokens from file if they have not already been loaded
+    # 2. Try to refresh access token with server using refresh token
+    # 3. (Re-) authenticate with server using credentials
+
+    global access_token, refresh_token
+
+    # 1. try to load from file
+    if access_token is None:
+        access_token, refresh_token = __load_tokens_from_file()
```
Comment (perchrh) [inline src/pytailor/clients/auth.py:40]: Only the refresh token needs to be stored in a file, access token is good enough to keep in memory, only

File: src/pytailor/clients/auth.py [hunk 2]
```diff
@@ -1,15 +1,111 @@
...
+def __load_tokens_from_file():
+    if TOKENS_FILE_PATH.exists():
+        with open(TOKENS_FILE_PATH, "rb") as f:
+            return pickle.load(f)
+    else:
+        return None, None
+
+
+def __write_tokens_to_file(access_token, refresh_token):
+    with open(TOKENS_FILE_PATH, "wb") as f:
```
Comment (perchrh) [inline src/pytailor/clients/auth.py:68]: Will missing directories in TOKENS_FILE_PATH have to be created?

**Comment by perchrh** [comment]
Using generic OAUTH2 requests would be an improvement, or if using Cognito-specific code - the boto3 SDK

File: src/pytailor/clients/auth.py (hunk 2 above)
Comment (agrav) [reply src/pytailor/clients/auth.py:68]: fixed in this PR

File: src/pytailor/clients/auth.py (hunk 1 above)
Comment (agrav) [reply src/pytailor/clients/auth.py:40]: Good improvement, will be noted and implemented later

**Comment by agrav** [comment]
> Using generic OAUTH2 requests would be an improvement, or if using Cognito-specific code - the boto3 SDK

Good tip, an effort was done with [authlib](https://authlib.org/) but without success. Will be looked into later.

**Pull request merged**